src/Microsoft.DotNet.Wpf/src/PresentationCore/MS/internal/Interop/DWrite/IDWriteFactory.cs

```diff
@@ -0,0 +1,58 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.CompilerServices;
+
+namespace MS.Internal.Interop.DWrite
+{
+    internal unsafe struct IDWriteFactory : IUnknown
```

Comment:
I would not try to replicate COM inheritance here. If you look how RCW/CCW with ComWrapppers implemented [in runtime](https://github.com/dotnet/runtime/pull/63969/files#diff-24e48862e3b82f52e7fa04f22700b1c976a012bfeb08a246406f4e5ec579699bR53), or even [more lean](https://github.com/dotnet/winforms/pull/6685/files#diff-9e75e0c456be18af94a8953ab771dd422b5ca1836345043a1f9212fce8f02f93R13) in WinForms.

For example `IUnknown` interface not needed, since you can use regular casts instead of QueryInterface and `Marshal.AddRef(IntPtr)`/`Marshal.Release(IntPtr)`

Reply:
I would prefer to leave it like this. I try to stay closer to the code generated by ClangSharp which maps the header files to C#. This offers some kind of type safety when using generics (Like [here](https://github.com/dotnet/wpf/pull/6171/files#diff-3945071e1113941167d91a8269971018f18b75ca6457fda26b6ac9baabafa1d3R24)). I also try to avoid using `Marshal.AddRef(IntPtr)/Marshal.Release(IntPtr)` because we can just call the methods implemented in the struct or in `IUnknown`.  The drawback is that the code is slightly larger but I think the type safety and cleaner code (In my opinion) is worth it.